mem3_helper.py

```diff
@@ -23,12 +23,29 @@ def construct_service_record():
     max_tries=10
 )
 def discover_peers(service_record):
-    print ('Resolving SRV record', service_record)
-    answers = dns.resolver.query(service_record, 'SRV')
-    # Erlang requires that we drop the trailing period from the absolute DNS
-    # name to form the hostname used for the Erlang node. This feels hacky
-    # but not sure of a more official answer
-    return [rdata.target.to_text()[:-1] for rdata in answers]
+    expected_peers_count = os.getenv('COUCHDB_CLUSTER_SIZE')
+    if expected_peers_count:
+        print('Expecting', expected_peers_count, 'peers...')
+    else:
+        print('Looks like COUCHDB_CLUSTER_SIZE is not set, will not wait for DNS...')
+    peers_count = 0
+    while str(peers_count) != expected_peers_count:
```

Comment:
It seems that this script is executed every time the pods are recreated, perhaps a limit number of tries could help to avoid unexpected behaviors hours later of the creation of the pods.

Reply:
Yeah, I was also thinking about this. Looks like the proper approach would be to utilize existing backoff decorator.